### PR TITLE
implement usage of aws_endpoint_url

### PR DESCRIPTION
**Pull request by pinzon**
## Motivation 
This PR optimizes the plugin by utilizing the AWS_ENDPOINT_URL environment variable as per the readme instructions. All default configuration values now derive from this variable, with a default value set to https://localhost.localstack.cloud:4566.

## Notes
Currently there are three conflicting ways the protocol, hostname and port are set up:
- env variables: USE_SSL, LOCALSTACK_HOSTNAME, and EDGE_PORT
- serverless config settings: edgeport and hostname
- env var AWS_ENDPOINT_URL

I propose that in the future it should only support AWS_ENDPOINT_URL env variable and also a new config attribute awsEndpointUrl.

### CODE REVIEW

**Review by steffyP — COMMENTED** [review]
Thanks for addressing the missing implementation, and prepare the code for the long announced deprecation of envs `EDGE_PORT`, `LOCALSTACK_HOSTNAME` and `USE_SSL`. 🚀 

One thing I realized though: in the README we state that the default value for `AWS_ENDPOINT_URL` will be `http://localhost:4566`, but now it actually is `https://localhost.localstack.cloud:4566`. 

I wonder if this could be an issue as we currently have a fallback for an issue with IPv6 address resolution in the code that checks for hostname `localhost` only:
https://github.com/localstack/serverless-localstack/blob/04f71f8ea8bee40a34e4713d8bb440982030cec2/src/index.js#L724-L739

So we may need to add `localhost.localstack.cloud` here as well. /cc @joe4dev could we get your input here? 🙏

File: volume/cache/machine.json
[the remark this answers is not in the document]
Comment (pinzon) [reply volume/cache/machine.json:1]: Done :+1:

**Comment by joe4dev** [comment]
> So we may need to add `localhost.localstack.cloud` here as well. /cc @joe4dev could we get your input here? 🙏

✅  This should not be necessary because `localhost.localstack.cloud` always resolves to IPv4 (at least with our current DNS entry). That's what I get on macOS:

```
nslookup localhost.localstack.cloud
Server:		1.1.1.1
Address:	1.1.1.1#53

Non-authoritative answer:
Name:	localhost.localstack.cloud
Address: 127.0.0.1
```


⚠️ However, I think changing the default to `localhost.localstack.cloud` is a breaking change and should not be done in isolation because it breaks in environments with DNS rebind protection => let me clarify in a review ...

**Comment by steffyP** [comment]
Thanks for your input @joe4dev 🙏 

@pinzon: I would suggest we set the `DEFAULT_AWS_ENDPOINT_URL` to `http://localhost:4566` (like it's also documented in the README), and revisit this when we do the next major release. Then we can also try to adjust this across integrations like @joe4dev suggested.

Regarding the code formatting: I think we can do this in a follow-up PR, it would indeed be nice to have formatting rules, and a pre-commit hook 😄

**Pull request merged**